cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs

```diff
@@ -3181,9 +3184,24 @@ pMaxTransactionSize =
       , Opt.help "Maximum transaction size."
       ]
 
+-- | @wordReader@ is a reader for a word of type @a@. When it fails
+-- parsing, it provides a nice error message. This custom reader is needed
+-- to avoid the overflow issues of 'Opt.auto' described in https://github.com/IntersectMBO/cardano-cli/issues/860.
+wordReader :: forall a. (Typeable a, Integral a, Bits a) => ReadM a
```

Comment:
We should abandon `read` entirely. Have a look at `readerFromParsecParser`.

Reply:
I think what @Jimbo4350 meant is that we should use parsec to have more flexible number parsing. I think this way you'll avoid writing validation for the numbers from scratch (`naturalToWordMaybe >=> toIntegralSized` in your case). See [`parseLovelace`](https://github.com/IntersectMBO/cardano-cli/blob/aa389b09afaebf09403072a05b09dd69872a917c/cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs#L514) for example. 

See also: https://hackage.haskell.org/package/parsec-3.1.17.0/docs/Text-Parsec-Token.html#t:GenTokenParser

Reply:
OK I see :+1: I did a `Parsec`-based implementation in the last commit.

Reply:
A better name would be `integralReader`.